src/lib/Search/PagerContentToDataMapperAbstract.php

```diff
@@ -0,0 +1,136 @@
+        UserService $userService,
+        UserLanguagePreferenceProviderInterface $userLanguagePreferenceProvider,
+        LanguageService $languageService
+    ) {
+        $this->contentTypeService = $contentTypeService;
+        $this->userService = $userService;
+        $this->userLanguagePreferenceProvider = $userLanguagePreferenceProvider;
+        $this->translationHelper = $translationHelper;
+        $this->languageService = $languageService;
+    }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\Content $content
+     * @param bool $filterDisabled
+     *
+     * @return \eZ\Publish\API\Repository\Values\Content\Language[]
+     */
+    protected function getAvailableTranslations(
+        Content $content,
+        bool $filterDisabled = false
+    ): iterable {
+        $availableTranslationsLanguages = $this->languageService->loadLanguageListByCode(
+            $content->versionInfo->languageCodes
+        );
+
+        if (false === $filterDisabled) {
+            return $availableTranslationsLanguages;
+        }
+
+        return array_filter(
+            $availableTranslationsLanguages,
+            (static function (Language $language): bool {
+                return $language->enabled;
+            })
+        );
+    }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\Content $content
+     *
+     * @return bool
+     */
+    protected function isContentIsUser(Content $content): bool
+    {
+        return (new ContentIsUser($this->userService))->isSatisfiedBy($content);
+    }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\ContentInfo $contentInfo
+     *
+     * @return \eZ\Publish\API\Repository\Values\User\User|null
+     */
+    protected function getContributor(ContentInfo $contentInfo): ?User
+    {
+        $userExists = (new UserExists($this->userService))->isSatisfiedBy($contentInfo->ownerId);
+
+        if (false === $userExists) {
+            return null;
+        }
+
+        return $this->userService->loadUser($contentInfo->ownerId);
+    }
+
+    /**
+     * @param array $data
+     * @param array $contentTypeIds
+     */
+    protected function setTranslatedContentTypesNames(array &$data, array $contentTypeIds): void
+    {
+        // load list of Content Types with proper translated names
+        $contentTypes = $this->contentTypeService->loadContentTypeList(
+            array_unique($contentTypeIds),
+            $this->userLanguagePreferenceProvider->getPreferredLanguages()
+        );
+
+        foreach ($data as $idx => $item) {
+            /** @var \eZ\Publish\API\Repository\Values\ContentType\ContentType $contentType */
+            // get content type from bulk-loaded list or fallback to lazy loaded one if not present
+            $contentType = isset($contentTypes[$item['contentTypeId']])
+                ? $contentTypes[$item['contentTypeId']]
+                : $item['content']->getContentType();
```

Comment:
It could be simplified to `$contentTypes[$item['contentTypeId']] ?? $item['content']->getContentType()`. To improve readability even more you can all extract `$item['contentTypeId']` to variable.